ldk/javascript/src/promisify.ts

```diff
@@ -0,0 +1,91 @@
+import { Cancellable } from './cancellable';
+
+export function promisify<T>(arg: OliveHelps.Readable<T>): Promise<T> {
+  return new Promise((resolve, reject) => {
+    try {
+      arg((cb: T) => resolve(cb));
+    } catch (e) {
+      console.error(e);
+      reject(e);
+    }
+  });
+}
+
+export function promisifyWithParam<TParam, TOut>(
+  param: TParam,
+  arg: OliveHelps.ReadableWithParam<TParam, TOut>,
+): Promise<TOut> {
+  return new Promise((resolve, reject) => {
+    try {
+      arg(param, (cb: TOut) => resolve(cb));
+    } catch (e) {
+      console.error(e);
+      reject(e);
+    }
+  });
+}
+
+export function promisifyWithTwoParams<TParam1, TParam2, TOut>(
+  param: TParam1,
+  param2: TParam2,
+  arg: OliveHelps.ReadableWithTwoParams<TParam1, TParam2, TOut>,
+): Promise<TOut> {
+  return new Promise((resolve, reject) => {
+    try {
+      arg(param, param2, (cb: TOut) => resolve(cb));
+    } catch (e) {
+      console.error(e);
+      reject(e);
+    }
+  });
+}
+
+export function promisifyWithFourParams<TP1, TP2, TP3, TP4, TOut>(
+  p1: TP1,
+  p2: TP2,
+  p3: TP3,
+  p4: TP4,
+  arg: OliveHelps.ReadableWithFourParams<TP1, TP2, TP3, TP4, TOut>,
+): Promise<TOut> {
```

Comment:
Is there a way to have arguments grouped, like: ...args: string[]?

Reply:
Yeah I wasn't a big fan of that function either.

I think we might not be able to do it easily. 

So... we can't just do a straight splat without sacrificing the type safety of the `OliveHelps.ReadableWithFourParams` I don't think. I think it might've been different if we had established the pattern that the callbacks would always be first.

Reply:
Only way I can think of is some sort of builder pattern: `promisify<T>()` function, which returns itself and then include another function `withParam<TParam>(arg: TParam)` which also returns a reference to that original `promisify<T>()` function. Finally a `.get()` or `.build()` on the end which returns `Promise<TOut>`.

Not sure this makes things more clear or not.

Reply:
Yeah, perhaps if we think of something better we could always come back to it.